File: 1260-shift-2d-grid/1260-shift-2d-grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        return solve(grid, k);
    }
private:
    vector<vector<int>> solve(vector<vector<int>>& grid, int k) {
        int n = grid.size(), m = grid[0].size();
        
        vector<vector<int>> ans(n, vector<int>(m, 0));
        
        for(int i = 0; i < n; ++i) {
            for(int j = 0; j < m; ++j) {
                int x = (i + (j + k) / m) % n;
                int y = (j + k) % m;
                
                ans[x][y] = grid[i][j];
            }
        }        
        return ans;
    }
};

File: 1260-shift-2d-grid/1260-shift-2d-grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        n = len(grid)
        m = len(grid[0])
        
        ans = [[0] * m for _ in range(n)]
        
        for i in range(n):
            for j in range(m):
                x = (i + (j + k) // m) % n
                y = (j + k) % m
                
                ans[x][y] = grid[i][j]
                
        return ans